play-json/shared/src/main/scala/play/api/libs/json/Json.scala

```diff
@@ -217,10 +217,15 @@ object Json extends JsonFacade {
 
   implicit def toJsFieldJsValueWrapper[T](field: T)(implicit w: Writes[T]): JsValueWrapper = JsValueWrapperImpl(w.writes(field))
 
-  def obj(fields: (String, JsValueWrapper)*): JsObject = JsObject(fields.map(f => (f._1, f._2.asInstanceOf[JsValueWrapperImpl].field)))
+  def obj(fields: (String, JsValueWrapper)*): JsObject = {
+    if (fields.isEmpty) JsObject.empty
```

Comment:
Improves? - yes,
significantly ? - no
There is another commit making `.empty` call to refer a constant value.

Reply:
If it doesn't significantly improve, considering the extra verbosity I'm not sure such change is welcome.

Reply:
@cchantep hm strange, I'm eager to put more effort in this and do as proper pull request with more places changes in similar way, I mean if we can return constants and not allocate things - we do so. It for sure will increase lines of code. However I'd avoid spending time on things that is rejected because of `extra verbosity`

And imho `extra verbosity` is bad argument against improvements...

Reply:
@t3hnar try [jsoniter-scala](https://github.com/plokhotnyuk/jsoniter-scala) it is much safe and efficient.  Usually it outperform play-json in ~10x times, will ~100x less allocations.

Reply:
@plokhotnyuk I don't think that's the place for such discussion

Reply:
@t3hnar verbosity without knowledgable improvement is a argument for me